src/main/java/com/zendesk/maxwell/replication/MaxwellReplicator.java

```diff
@@ -114,6 +116,28 @@ private void ensureReplicatorThread() throws Exception {
 	public void startReplicator() throws Exception {
 		try {
 			this.replicator.start();
+
+			replicator.getBinlogParser().addParserListener(new BinlogParserListener() {
+				@Override
+				public void onStart(BinlogParser parser) {}
+
+				@Override
+				public void onException(BinlogParser parser, Exception exception) {
+					LOGGER.error("Binlog parse error: " + exception.getLocalizedMessage());
+				}
+
+				@Override
+				public void onStop(BinlogParser parser) {
+					queue.clear();
+					try {
+						//wait a bit here for new messages to be pushed to the queue since the queue might be blocked before
+						Thread.sleep(2000);
+					} catch (InterruptedException e) {
+					}
+					//clear again in case some messages were pushed just now
+					queue.clear();
+				}
+			});
```

Comment:
This sleeping business isn't ideal, but it's better than what we've got. Hopefully we can deprecate this connector soon anyway.